src/table/table.scss

```diff
@@ -122,265 +249,341 @@
+        @include themed {
+          box-shadow: inset $iui-xxs 0 0 0 t(iui-icons-color-warning);
+        }
 
-  .iui-tables-message-container {
-    @include themed {
-      color: t(iui-text-color-muted);
-      background-color: t(iui-color-background-2);
-    }
-    display: flex;
-    justify-content: center;
-    height: 100%;
-    padding: $iui-xl;
-  }
+        *::selection {
+          @include themed {
+            background-color: rgba(
+              t(iui-color-foreground-warning-rgb),
+              t(iui-opacity-4)
+            );
+          }
+        }
+      }
 
-  .iui-tables-paginator {
-    @include themed {
-      background-color: t(iui-color-background-3);
+      .iui-status {
+        @include themed {
+          fill: t(iui-icons-color-warning);
+        }
+      }
     }
-    user-select: none;
 
-    &:hover {
-      @include themed {
-        background-color: t(iui-color-background-3) !important;
+    &.iui-negative {
+      &,
+      + .iui-expanded-content {
+        @include themed {
+          box-shadow: inset $iui-xxs 0 0 0 t(iui-icons-color-negative);
+        }
+
+        *::selection {
+          @include themed {
+            background-color: rgba(
+              t(iui-color-foreground-negative-rgb),
+              t(iui-opacity-4)
+            );
+          }
+        }
       }
-    }
 
-    > .iui-content {
-      @include themed {
-        border-bottom: solid 1px t(iui-color-background-4);
+      .iui-status {
+        @include themed {
+          fill: t(iui-icons-color-negative);
+        }
       }
-      width: 100%;
-      display: flex;
-      justify-content: space-around;
-      align-items: center;
-      padding: $iui-table-density;
-      flex: 1;
+    }
 
-      > .iui-full-line {
-        flex: 1;
+    &.iui-expanded {
+      .iui-expander > .iui-icon {
+        transform: rotate(90deg);
       }
 
-      > .iui-controls {
+      + .iui-expanded-content {
         @include themed {
-          color: t(iui-text-color);
+          border-top: solid 1px t(iui-color-background-1);
         }
-        width: 300px;
-        display: flex;
-        justify-content: center;
-        align-items: center;
-        flex: 1;
+      }
+    }
+    // #endregion Selection
+  }
 
-        > a {
-          @include iui-anchor;
-          margin: 0 $iui-s;
-        }
+  // Empty & loading states
+  &.iui-message {
+    text-align: center;
+    height: 100%;
+    padding: $iui-xl;
+    @include themed {
+      color: t(iui-text-color-muted);
+      background-color: t(iui-color-background-2);
+    }
+  }
+}
 
-        > .iui-ellipsis {
-          margin: 0 $iui-s;
-        }
+@mixin iui-table-cell {
+  align-items: center;
+  display: flex;
+  flex-grow: 1;
+  min-width: $iui-l * 6;
+  min-height: $iui-baseline * 5;
+  padding: 0 $iui-m;
+  overflow: hidden;
+  list-style: none;
+  flex-basis: $iui-l * 6;
+
+  > svg {
+    width: $iui-icons-default;
+    height: $iui-icons-default;
```

Comment:
🤔

Reply:
Switched this to use `.iui-icon` instead of `svg`. Am not using `@include iui-icons-default;` because it would override the colors of the filetype icons.

Reply:
`.iui-icon` still seems too generic. Remind me again, why this is necessary

Reply:
I think we should define a variable that would imply all the possibilities for svg/iui-icon

Reply:
amm, so we still end up duplicating rules...
I do understand the need of flex-shrink, but size is duplicated, like in this example:
![image](https://user-images.githubusercontent.com/81580355/120780759-174e2a00-c531-11eb-87fe-f27c1b5233cb.png)

I guess we might need to think about structuring this whole thing.. Thinking from componentization side:
Have different types of Cell (separate class/mixin).
Now its a little bit of mess in the rules in this file..